File: frontend/src/types/guidedReview.ts
export type GuidedReviewStatus = 'pending' | 'in_progress' | 'completed' | 'skipped'

export type GuidedReview = {
  id: string
  user_id: string
  review_date: string
  status: GuidedReviewStatus
  completed_at?: string | null
  created_at: string
}

export type GuidedReviewItem = {
  id: string
  review_id: string
  trade_id?: string | null
  bundle_key?: string | null
  symbol: string
  side?: string | null
  pnl?: number | null
  trade_count: number
  intent?: string | null
  emotions?: string[] | null
  pattern_match?: string | null
  memo?: string | null
  order_index: number
  created_at: string
}

export type UserStreak = {
  user_id: string
  current_streak: number
  longest_streak: number
  last_review_date?: string | null
  updated_at: string
}

export type TodayResponse = {
  review: GuidedReview
  items: GuidedReviewItem[]
}

export type SubmitItemPayload = {
  intent: string
  emotions: string[]
  pattern_match: string
  memo: string
}

export type CompleteResponse = {
  ok: boolean
  streak: UserStreak
}

export const INTENT_OPTIONS = [
  { value: 'technical_signal', label: '기술적 신호' },
  { value: 'news_event', label: '뉴스/이벤트' },
  { value: 'emotional', label: '감정적 판단' },
  { value: 'planned_regular', label: '계획된 매매' },
  { value: 'other', label: '기타' },
] as const

export const EMOTION_OPTIONS = [
  { value: 'confident', label: '확신' },
  { value: 'half_doubtful', label: '반신반의' },
  { value: 'anxious', label: '불안' },
  { value: 'excited', label: '흥분' },
  { value: 'calm', label: '평온' },
  { value: 'nervous', label: '초조' },
  { value: 'fomo', label: 'FOMO' },
  { value: 'revenge_trade', label: '복수매매' },
  { value: 'as_planned', label: '계획대로' },
] as const

export const PATTERN_OPTIONS = [
  { value: 'same_decision', label: '같은 판단' },
  { value: 'adjust_timing', label: '타이밍 조절' },
  { value: 'reduce_size', label: '사이즈 축소' },
  { value: 'would_not_trade', label: '안 하겠다' },
  { value: 'change_sl_tp', label: 'SL/TP 변경' },
] as const

File: backend/internal/domain/entities/guided_review.go
package entities

import (
	"encoding/json"
	"time"

	"github.com/google/uuid"
)

// GuidedReview status constants
const (
	GuidedReviewStatusPending    = "pending"
	GuidedReviewStatusInProgress = "in_progress"
	GuidedReviewStatusCompleted  = "completed"
	GuidedReviewStatusSkipped    = "skipped"
)

// Intent constants (Layer 1)
const (
	IntentTechnicalSignal = "technical_signal"
	IntentNewsEvent       = "news_event"
	IntentEmotional       = "emotional"
	IntentPlannedRegular  = "planned_regular"
	IntentOther           = "other"
)

// Emotion constants (Layer 2)
const (
	EmotionGRConfident    = "confident"
	EmotionGRHalfDoubtful = "half_doubtful"
	EmotionGRAnxious      = "anxious"
	EmotionGRExcited      = "excited"
	EmotionGRCalm         = "calm"
	EmotionGRNervous      = "nervous"
	EmotionGRFomo         = "fomo"
	EmotionGRRevengeTrade = "revenge_trade"
	EmotionGRAsPlanned    = "as_planned"
)

// PatternMatch constants (Layer 3)
const (
	PatternSameDecision  = "same_decision"
	PatternAdjustTiming  = "adjust_timing"
	PatternReduceSize    = "reduce_size"
	PatternWouldNotTrade = "would_not_trade"
	PatternChangeSlTp    = "change_sl_tp"
)

type GuidedReview struct {
	ID          uuid.UUID  `json:"id"`
	UserID      uuid.UUID  `json:"user_id"`
	ReviewDate  string     `json:"review_date"`
	Status      string     `json:"status"`
	CompletedAt *time.Time `json:"completed_at,omitempty"`
	CreatedAt   time.Time  `json:"created_at"`
}

type GuidedReviewItem struct {
	ID           uuid.UUID       `json:"id"`
	ReviewID     uuid.UUID       `json:"review_id"`
	TradeID      *uuid.UUID      `json:"trade_id,omitempty"`
	BundleKey    *string         `json:"bundle_key,omitempty"`
	Symbol       string          `json:"symbol"`
	Side         *string         `json:"side,omitempty"`
	PnL          *float64        `json:"pnl,omitempty"`
	TradeCount   int             `json:"trade_count"`
	Intent       *string         `json:"intent,omitempty"`
	Emotions     json.RawMessage `json:"emotions,omitempty"`
	PatternMatch *string         `json:"pattern_match,omitempty"`
	Memo         *string         `json:"memo,omitempty"`
	OrderIndex   int             `json:"order_index"`
	CreatedAt    time.Time       `json:"created_at"`
}

type UserStreak struct {
	UserID         uuid.UUID `json:"user_id"`
	CurrentStreak  int       `json:"current_streak"`
	LongestStreak  int       `json:"longest_streak"`
	LastReviewDate *string   `json:"last_review_date,omitempty"`
	UpdatedAt      time.Time `json:"updated_at"`
}
